modules/tinymce/src/plugins/media/test/ts/browser/SubmitTest.ts

```diff
@@ -1,117 +1,165 @@
-    await pTestManualEmbedContentSubmit(editor, customEmbed, customEmbed);
-    editor.options.unset('media_url_resolver');
-    await pTestEmbedUnchangedAfterOpenCloseDialog(editor, customEmbed);
-  });
-
-  it('TBA: Set audio_template_callback and embed content, submit dialog and assert content', async () => {
-    // Audio should be any mp3 link to trigger callback
-    const editor = hook.editor();
-    const audioTemplateCallback = Fun.constant('<audio id="template" controls="controls"><source src="https://assets.mixkit.co/music/preview/mixkit-tech-house-vibes-130.mp3"></audio>');
-    const expected = '<p><audio id="template" controls="controls"><source src="https://assets.mixkit.co/music/preview/mixkit-tech-house-vibes-130.mp3"></audio></p>';
-    await pTestTemplateCallbackContentSubmit(editor, 'audio_template_callback', audioTemplateCallback, 'https://assets.mixkit.co/music/preview/mixkit-tech-house-vibes-130.mp3', expected);
-    await pTestEmbedUnchangedAfterOpenCloseDialog(editor, expected);
-  });
-
-  it('TBA: Set video_template_callback and embed content, submit dialog and assert content', async () => {
-    // Video should be any mp4 link to trigger callback
-    const editor = hook.editor();
-    const videoTemplateCallback = Fun.constant('<video id="template" controls="controls" width="500" height="300" ><source src="https://i.imgur.com/Tu4e5WX.mp4"></video>');
-    const expected = '<p><video id="template" controls="controls" width="500" height="300"><source src="https://i.imgur.com/Tu4e5WX.mp4"></video></p>';
-    await pTestTemplateCallbackContentSubmit(editor, 'video_template_callback', videoTemplateCallback, 'https://i.imgur.com/Tu4e5WX.mp4', expected);
-    await pTestEmbedUnchangedAfterOpenCloseDialog(editor, expected);
-  });
-
-  it('TINY-8684: Set iframe_template_callback and embed content, submit dialog and assert content', async () => {
-    // Any youtube link triggers iframe callback
-    const editor = hook.editor();
-    const iframeTemplateCallback = Fun.constant('<iframe id="template" title="testcallback" src="https://www.youtube.com/embed/IcgmSRJHu_8" width="500" height="300"></iframe>');
-    const expected = '<p><iframe id="template" title="testcallback" src="https://www.youtube.com/embed/IcgmSRJHu_8" width="500" height="300"></iframe></p>';
-    await pTestTemplateCallbackContentSubmit(editor, 'iframe_template_callback', iframeTemplateCallback, 'https://www.youtube.com/embed/IcgmSRJHu_8', expected);
-    await pTestEmbedUnchangedAfterOpenCloseDialog(editor, expected);
+  context('youtube embed', () => {
+    const hook = TinyHooks.bddSetup<Editor>({
+      plugins: [ 'media' ],
+      toolbar: 'media',
+      base_url: '/project/tinymce/js/tinymce',
+      setup: (editor: Editor) => {
+        editor.on('PreInit', () => {
+          const converter = (nodes: AstNode[]): void => {
+            Arr.each(nodes, (node) => {
+              const shimNode = new tinymce.html.Node('span', 1);
```

Comment:
We should be able to use the AstNode class directly that comes from `import AstNode from 'tinymce/core/api/html/Node';` 

That way we can avoid the `import { tinymce } from 'tinymce/core/api/Tinymce';`

This change can be made in a few other places too it seems.

```suggestion
              const shimNode = new AstNode('span', 1);
```

Reply:
Changed from `tinymce.html.Node` to `AstNode`.